crates/autopilot/src/solvable_orders.rs

```diff
@@ -236,6 +249,29 @@ impl SolvableOrdersCache {
     pub fn last_update_time(&self) -> Instant {
         self.cache.lock().unwrap().orders.update_time
     }
+
+    /// Filter out limit orders which are far enough outside the estimated native token price.
+    fn filter_mispriced_limit_orders(
+        &self,
+        mut orders: Vec<Order>,
+        prices: &BTreeMap<H160, U256>,
+    ) -> Vec<Order> {
+        orders.retain(|order| {
+            if order.metadata.class != OrderClass::Limit {
+                return true;
+            }
+
+            // Convert the sell and buy price to the native token (ETH) and make sure that sell
+            // with the surplus fee is higher than buy with the configurable price factor.
+            let sell_native = (order.data.sell_amount + order.metadata.surplus_fee)
+                * prices.get(&order.data.sell_token).unwrap();
+            let buy_native = order.data.buy_amount * prices.get(&order.data.buy_token).unwrap();
+            let sell_native = u256_to_big_decimal(&sell_native);
+            let buy_native = u256_to_big_decimal(&buy_native);
+            sell_native >= buy_native * self.limit_order_price_factor.clone()
```

Comment:
Nit: Sometimes its hard to debug why an order wasn't included in a batch. Can you add a log message including the order UID for when it gets filtered out?